intl_translation_format/lib/test/format_test.dart

```diff
@@ -1,174 +1,85 @@
 import 'package:intl_translation_format/intl_translation_format.dart';
-import 'package:intl_translation_format/test/test_mock.dart' as mock;
 import 'package:test/test.dart';
 
-dynamic expectMessages(
+/// Mock Template used for testing
+///
+/// The messages are added programatically in the constructor
+///
+class MockTemplate extends TranslationTemplate {
+  MockTemplate(
+    String projectName,
+    this.messages, {
+    String locale = 'en',
+  }) : super(
+          projectName,
+          locale: locale,
+        );
+  @override
+  final Map<String, MainMessage> messages;
+}
+
+// Compares two message with their icu string version.
+dynamic expectMessage(Message message, Message expected) {
+  return expect(
+    messageToIcuString(message),
+    messageToIcuString(expected),
+  );
+}
+
+/// Compares a file content with the messages that are expected
+/// after parsing the file with the indicated [format]
+dynamic expectFormatParsing(
   String content,
   MonoLingualFormat format, {
-  Map<String, String> messages = const {},
-  List<String> arguments = const [],
+  List<MainMessage> messages = const [],
 }) {
   final result = format.parseFile(content);
-  final MainMessage mainMessage = MainMessage()..arguments = arguments;
-  final parsed = result.messages.map(
-    (key, m) => MapEntry(
-      key,
-      messageToIcuString(m.message..parent = mainMessage),
-    ),
-  );
-  return expect(parsed, messages);
+
+  for (final translated in result.messages.values) {
+    final mainMessage = messages.firstWhere((e) => e.name == translated.id);
+    final translatedMessage = translated.message..parent = mainMessage;
+    expectMessage(translatedMessage, mainMessage);
+  }
 }
 
-dynamic expectContentForMessages(
+/// Compares a file content with the messages that are expected
+/// after parsing the file with the indicated [format]
+///
+/// While [expectFormatParsing] works with monolingual files,
+/// [expectMulipluralFormatParsing] allows to test formats that
+/// contains multiple languages
+///
+dynamic expectMulipluralFormatParsing(
   String content,
-  MonoLingualFormat format, {
-  Map<String, MainMessage> messages = const {},
-  List<String> arguments = const [],
+  MultiLingualFormat format, {
+  Map<String, List<MainMessage>> messages,
+  String defaultLocale = 'en',
 }) {
-  final template = mock.MockTemplate(
+  final result = format.parseFile(content, defaultLocale);
```

Comment:
Better name than "result"? It's not obvious what this is, and why it has multiple things, each of which is called "translatedForLocale". So maybe this is "allTranslations"?